### PR TITLE
Rename HealthKit observation related FeatureFlag

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1417
- Rename FeatureFlags.observeHealthKitForCurrentAppOnly to observeHealthKitSamplesFromOtherApps
- Rename OBSERVE_HEALTHKIT_FOR_CURRENT_APP_ONLY to OBSERVE_HEALTH_KIT_SAMPLES_FROM_OTHER_APPS_DISABLED
- Add observeHealthKitSamplesFromCurrentApp to HealthKitSampleStore
- Set default observeHealthKitSamplesFromOtherApps for various stores
- Preparation for CarbStore using Core Data as "source of truth"

Note: The primary reason for this change is future work for LOOP-1417 where the "source of truth" for Carb data will be in Core Data and not HealthKit. At this point, Loop-sourced data will no longer be read from HealthKit. This yields 4 states for reading data from HealthKit in HealthKitSampleStore and thus the need for two booleans:

1. read Loop data, read non-Loop data - DIY insulin, glucose
2. read Loop data, don't read non-Loop data - Tidepool insulin, glucose
3. don't read Loop data, read non-Loop data - DIY carbs
4. don't read Loop data, don't read non-Loop data - Tidepool carbs

Note: I'm breaking up LOOP-1417 into multiple smaller PRs (though the last one is still quite large).

### CODE REVIEW

File: LoopKit/HealthKitSampleStore.swift [hunk 1]
```diff
@@ -214,16 +219,40 @@ extension HealthKitSampleStore: HKSampleQueryTestable {
...
+// MARK: - Query
+extension HealthKitSampleStore {
+    internal func predicateForSamples(withStart startDate: Date?, end endDate: Date?, options: HKQueryOptions = []) -> NSPredicate? {
+        guard observeHealthKitSamplesFromCurrentApp || observeHealthKitSamplesFromOtherApps else {
+            return nil
+        }
+
+        var predicate = HKQuery.predicateForSamples(withStart: startDate, end: endDate, options: options)
+
+        if !observeHealthKitSamplesFromCurrentApp {
```
Comment (rickpasetto) [inline LoopKit/HealthKitSampleStore.swift:231]: Wow...I had to stare at this boolean logic a while to make sure it was all good.  Nice job!  

[nit] maybe a comment in here would help future readers?  Something like
```
// Exclude current app data, but include other app data
```
```
// Include current app data, exclude other app data
```
?

**Review by rickpasetto — APPROVED** [review]
LGTM 🦑

**Review by ps2 — APPROVED** [review]
LGTM!

File: LoopKit/HealthKitSampleStore.swift (hunk 1 above)
Comment (darinkrauss) [reply LoopKit/HealthKitSampleStore.swift:231]: I will put this in a subsequent PR.

**Pull request merged**